Vankeken 2008 subduction benchmark (#5259)
geodynamics/aspect: cookbooks/vankeken_subduction/doc/vankeken_subduction.md

# Van Keken 2008 corner flow recreation

> *This section was contributed by Daniel Douglas and Max Rudolph*.

Reviewer: I think you probably should also credit Cedric.

Author: Definitely agree!

> This cookbook recreates a 2D corner flow style subduction model as outlined in van Keken 2008.

Reviewer: Can you add a proper reference to the publication?

Since ASPECT uses a rectangular mesh by default, the geometry of this model combined with the
way that the velocities are constrained make using ASPECT's default mesh extremely unstable when
trying to solve for a continuous pressure field. To circumvent this issue, a custom mesh using

> quadrilaterals was designed that allows for a q2q1 discretization to become stable. The input

Reviewer: ```suggestion
quadrilaterals was designed that allows for a $Q_2\times Q_1$ discretization to become stable. The input
```

> files were modified only slightly from an existing PR #1864 made by Max Rudolph. Further efforts

Reviewer: Remove the reference to the old PR and its author here. Documentation should document the current state of a software, not its genesis or previous versions.

to confirm that this setup can more accurately reproduce the results of the 2008 benchmark should
be done to move this from the cookbook section to the benchmark section.

> The model prescribes a 5cm/yr at a subduction angle of 45 degrees convergence rate along the left boundary and the interface between the subducting plate and the overriding plate and mantle wedge. In the overriding plate, the velocity is constrained to be 0. The Stokes equations are solved in the mantle wedge, and the right and bottom boundaries are open.

Reviewer: ```suggestion
The model prescribes a 5cm/yr convergence rate at a subduction angle of 45 degrees along the left boundary and the interface between the subducting plate and the overriding plate and mantle wedge. In the overriding plate, the velocity is constrained to be 0. The Stokes equations are solved in the mantle wedge, and the right and bottom boundaries are open.
```

Reviewer: After this paragraph, can you add a few sentences that (i) describe what you observe when you run this scenario, and that (ii) references all figures you add?

```{figure-md} fig:vanKeken-pressure
<img src="pressure_mesh.png" alt="Screenshot"  width="100%"/>

> The pressure field in the corner of the mantle wedge. The pressure solution was solved using a continuous q1 discretization, which was made possible with the specific mesh design.

Reviewer: ```suggestion
The pressure field in the corner of the mantle wedge. The pressure solution was solved using a continuous $Q_1$ discretization, which was made possible with the specific mesh design.
```

> The half space cooling model prescribed on the left boundary is advected downdip at the prescribed convergence rate, creating the slab. The cold slab interacts with the mantle wedge, resulting in a classic corner flow temperature (figure 1), and velocity field (figure 2). The geometry of the model requires a very low dynamic pressure in the corner of the mantle wedge, and with the specialized mesh the pressure field was able to be solved for using a continuous $Q_1$ discretization, as shown in figure 2. The model proceeds until a steady state thermal structure is reached for the entire model domain.

Reviewer 2: ```suggestion
The half space cooling model prescribed on the left boundary is advected downdip at the prescribed convergence rate, creating the slab. The cold slab interacts with the mantle wedge, resulting in a classic corner flow temperature (figure 1) and velocity field (figure 2). The geometry of the model requires a very low dynamic pressure in the corner of the mantle wedge, and with the specialized mesh the pressure field was able to be solved for using a continuous $Q_1$ discretization, as shown in figure 2. The model proceeds until a steady state thermal structure is reached for the entire model domain.
```

> The temperature field of the subducting slab after ~100 Myr. A half space cooling model is applied to the left boundary which gets advected at a 45 degree angle at a rate of 5 cm/yr.

Reviewer 2: ```suggestion
The temperature field of the subducting slab after ~100 Myr. A half space cooling model is applied to the left boundary, which is advected at a 45 degree angle at a rate of 5 cm/yr within the subducting plate.
```

> The velocity field of the subducting slab after ~100 Myr. A half space cooling model is applied to the left boundary which gets advected at a 45 degree angle at a rate of 5 cm/yr.

Reviewer 2: ```suggestion
The velocity field of the subducting slab after ~100 Myr. which is advected at a 45 degree angle at a rate of 5 cm/yr within the subducting plate.
```

```

```{figure-md} fig:vanKeken-pressure
<img src="pressure_mesh.png" alt="Screenshot"  width="100%"/>

> The pressure field in the corner of the mantle wedge. The pressure solution was solved using a continuous $Q_1$ discretization, which was made possible with the specific mesh design.

Reviewer 2: ```suggestion
The pressure field in the corner of the mantle wedge. The pressure solution was solved using a continuous $Q_1$ discretization, which was made possible with the custom mesh design used for this cookbook.
```
